ansys_optical_automation/application/stack_convert_to_speos.py

```diff
@@ -0,0 +1,45 @@
+from opcode import stack_effect
+import os
+import tkinter as tk
+from tkinter import filedialog
+
+from ansys_optical_automation.post_process.dpf_stack import DpfStack
+
+def getfilename(extension, save=False):
+    """
+    Parameters
+    ----------
+    extension : str
+        containing the which file extension in *.ending format
+    save : Bool
+        option to define to open(default) or save. The default is False.
+
+    Returns
+    -------
+    str
+        string containing the selected file path
+    """
+    root = tk.Tk()
+    root.withdraw()
+    if not save:
+        file_path = filedialog.askopenfilename(filetypes=[("Stack data", extension)])
+    else:
+        file_path = filedialog.asksaveasfilename(filetypes=[("Stack data", extension)])
+        if not file_path.endswith(extension.strip("*")):
+            file_path += extension.strip("*")
+    return file_path
+
+
+def main():
+    stack_test_version = 222
+    stack_result_file = getfilename("*.ldf")
+    if not os.path.exists(stack_result_file):
+        msg = "Cannot find specified file: " + stack_result_file
+        raise TypeError(msg)
+    stack_dpf = DpfStack(stack_test_version)
+    stack_dpf.open_file(stack_result_file)
+    stack_dpf.convert_stack_to_speos()
+    print("Success: conversion of stack to Speos coating is done")
+
```

Comment:
```suggestion


```

Reply:
As suggested by the failed pre-commmit style automation check, please run "pre-commit run -a" at terminal to make sure following Python PEP 8 standard